app/models/declaration/submissions/SubmissionStatus.scala

```diff
@@ -15,136 +15,36 @@
  */
 
 package models.declaration.submissions
-
-import play.api.libs.json._
-
-sealed trait SubmissionStatus {
-  val fullCode: String
-}
-
-object SubmissionStatus {
-
-  implicit object StatusFormat extends Format[SubmissionStatus] {
-    def reads(status: JsValue): JsResult[SubmissionStatus] = status match {
-      case JsString(code) => JsSuccess(getStatusOrUnknown(code))
-      case _              => JsSuccess(UnknownStatus)
-    }
-
-    def writes(status: SubmissionStatus): JsValue = JsString(status.fullCode)
-  }
-
-  private val codesMap: Map[String, SubmissionStatus] = Map(
-    Pending.fullCode -> Pending,
-    RequestedCancellation.fullCode -> RequestedCancellation,
-    Accepted.fullCode -> Accepted,
-    Received.fullCode -> Received,
-    Rejected.fullCode -> Rejected,
-    UndergoingPhysicalCheck.fullCode -> UndergoingPhysicalCheck,
-    AdditionalDocumentsRequired.fullCode -> AdditionalDocumentsRequired,
-    Amended.fullCode -> Amended,
-    Released.fullCode -> Released,
-    Cleared.fullCode -> Cleared,
-    Cancelled.fullCode -> Cancelled,
-    CustomsPositionGranted.fullCode -> CustomsPositionGranted,
-    CustomsPositionDenied.fullCode -> CustomsPositionDenied,
-    GoodsHaveExitedTheCommunity.fullCode -> GoodsHaveExitedTheCommunity,
-    DeclarationHandledExternally.fullCode -> DeclarationHandledExternally,
-    AwaitingExitResults.fullCode -> AwaitingExitResults,
-    UnknownStatus.fullCode -> UnknownStatus
-  )
-
-  def retrieve(functionCode: String, nameCode: Option[String]): SubmissionStatus =
-    getStatusOrUnknown(functionCode + nameCode.getOrElse(""))
-
-  private def getStatusOrUnknown(searchKey: String): SubmissionStatus =
-    codesMap.get(searchKey) match {
-      case Some(status) => status
-      case None         => UnknownStatus
-    }
-
-  case object Pending extends SubmissionStatus {
-    val fullCode: String = "Pending"
-  }
-
-  case object RequestedCancellation extends SubmissionStatus {
-    val fullCode: String = "Cancellation Requested"
-
-    override def toString: String = "Cancellation Requested"
-  }
-
-  case object Accepted extends SubmissionStatus {
-    val fullCode: String = "01"
-  }
-
-  case object Received extends SubmissionStatus {
-    val fullCode: String = "02"
-  }
-
-  case object Rejected extends SubmissionStatus {
-    val fullCode: String = "03"
-  }
-
-  case object UndergoingPhysicalCheck extends SubmissionStatus {
-    val fullCode: String = "05"
-
-    override def toString(): String = "Undergoing Physical Check"
-  }
-
-  case object AdditionalDocumentsRequired extends SubmissionStatus {
-    val fullCode: String = "06"
-
-    override def toString(): String = "Additional Documents Required"
+import play.api.libs.json.Format
+import utils.EnumJson
+
+object SubmissionStatus extends Enumeration {
```

Comment:
I can't agree with this change. Case object and sealed trait it's the best approach to use with Scala for now.

Reply:
But you approved same change on BE.  And look at all that useless code I got rid of.

Reply:
I approved changed with removing function code and name code, not changes to change sealed traits to case objects